### PR TITLE
MMS functionality for version 1.x

### PR DESCRIPTION
**Pull request by CodyPChristian**
Updated Twilio.php to use different sending function from Twilio SDK. Previous implementation was limited to 160 characters.
Also implemented a MMS sending function.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+FORKED COPY
+This version has been forked from the original author. My changes as as follows:
+
+Master Branch - Was for Laravel 5 and was removed, this fork is only for Laravel 4.
+Updated Twilio.php to use different sending function from Twilio SDK. Previous implementation was limited to 160 characters.
+Also implemented a MMS sending function.
+
+Original: https://github.com/aloha/laravel-twilio
+
+
```
Comment (hannesvdvreken) [inline README.md]: :confused:

File: src/Aloha/Twilio/Twilio.php [hunk 2]
```diff
@@ -13,13 +13,24 @@ public function __construct($config) {
...
         // Send SMS via Twilio SDK
-        return $twilio->account->sms_messages->create(
+        return $twilio->account->messages->sendMessage(
             is_null($from) ? $this->config['from'] : $from,
             $to,
             $message
         );
     }
 
+    public function messageMMS($to, $message, $media, $from=null) {
```
Comment (hannesvdvreken) [inline src/Aloha/Twilio/Twilio.php]: have you checked https://github.com/aloha/laravel-twilio/blob/master/src/TwilioInterface.php#L23?

File: README.md (hunk 1 above)
Comment (CodyPChristian) [inline README.md]: That occurred from my copy, I can remove that from the pull request.

File: src/Aloha/Twilio/Twilio.php (hunk 2 above)
Comment (CodyPChristian) [inline src/Aloha/Twilio/Twilio.php]: Yes but that is in the Laravel 5 branch, this was not available in the Laravel 4 version that we use.

Comment (hannesvdvreken) [inline src/Aloha/Twilio/Twilio.php]: do you mind renaming it? I think the other name is better. And allows more easy upgrade from 1.x to 2.x later on.

Comment (CodyPChristian) [inline src/Aloha/Twilio/Twilio.php]: Hey, yes not a problem, I'll submit a new request with the fixed files and include a new test for MMS as well.

**Comment by CodyPChristian** [comment]
New commits available that contain the requested changes

File: README.md [hunk 3]
```diff
@@ -48,6 +49,16 @@ Twilio::message('+18085551212', 'Pink Elephants and Happy Rainbows');
...
 ```
 
+Sending a MMS Message
+
+```php
+<?php
+
+Twilio::messageWithMedia('+18085551212', 'Pink Elephants and Happy Rainbows', array('http://placehold.it/200x200'));
+
+?>
```
Comment (hannesvdvreken) [inline README.md]: could you remove the closing tags of all examples, please?

File: src/Aloha/Twilio/Commands/TwilioMmsCommand.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+        $this->line($text);
+
+        // Grab the image option if specified
+        $image_url = $this->option('image_url');
+
+        // If we havent specified a message, setup a default one
+        if(is_null($image_url)) {
+            $image_url = "http://placehold.it/200x200";
+        }
```
Comment (hannesvdvreken) [inline src/Aloha/Twilio/Commands/TwilioMmsCommand.php]: An MMS without any media, isn't that possible?

File: src/Aloha/Twilio/Commands/TwilioMmsCommand.php [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+        // If we havent specified a message, setup a default one
+        if(is_null($text)) {
+            $text = "This is a test message sent from the artisan console";
+        }
+
+        $this->line($text);
+
+        // Grab the image option if specified
+        $image_url = $this->option('image_url');
```
Comment (hannesvdvreken) [inline src/Aloha/Twilio/Commands/TwilioMmsCommand.php]: variable names should be `camelCase`d

File: src/Aloha/Twilio/Commands/TwilioMmsCommand.php [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
```
Comment (hannesvdvreken) [inline src/Aloha/Twilio/Commands/TwilioMmsCommand.php]: redundant.

File: src/Aloha/Twilio/Commands/TwilioMmsCommand.php [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+     * @var string
+     */
+    protected $name = 'twilio:mms';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Twilio command to test Twilio API Integration.';
```
Comment (hannesvdvreken) [inline src/Aloha/Twilio/Commands/TwilioMmsCommand.php]: 'to test Twilio API Integration', not really, is it?

File: src/Aloha/Twilio/Commands/TwilioMmsCommand.php [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        $this->line('Sending MMS via Twilio to: '.$this->argument('phone'));
+
+        // Grab the text option if specified
+        $text = $this->option('text');
+
+        // If we havent specified a message, setup a default one
+        if(is_null($text)) {
+            $text = "This is a test message sent from the artisan console";
+        }
```
Comment (hannesvdvreken) [inline src/Aloha/Twilio/Commands/TwilioMmsCommand.php]: Same here, just make the argument required.

**Comment by hannesvdvreken** [comment]
After this PR, I'd love to see a PR to the other branch with a console command for sending MMS, and the tests as well. :ok_hand:

File: README.md (hunk 3 above)
Comment (CodyPChristian) [inline README.md]: Sure, just copied what you had done. It'll be in my next commit.

File: src/Aloha/Twilio/Commands/TwilioMmsCommand.php (hunk 4 above)
Comment (CodyPChristian) [inline src/Aloha/Twilio/Commands/TwilioMmsCommand.php]: Don't believe so, sending a MMS without media would be pointless and cost you a higher rate vs sending a text-only SMS.

File: src/Aloha/Twilio/Commands/TwilioMmsCommand.php (hunk 5 above)
Comment (CodyPChristian) [inline src/Aloha/Twilio/Commands/TwilioMmsCommand.php]: No problem, I'll update it

File: src/Aloha/Twilio/Commands/TwilioMmsCommand.php (hunk 6 above)
Comment (CodyPChristian) [inline src/Aloha/Twilio/Commands/TwilioMmsCommand.php]: You have this same block in the other two command files as well. My new commit has updated the description to match the call.

File: src/Aloha/Twilio/Commands/TwilioMmsCommand.php (hunk 8 above)
Comment (CodyPChristian) [inline src/Aloha/Twilio/Commands/TwilioMmsCommand.php]: Text is optional, imageUrl is required, updated code to reflect this.

**Comment by CodyPChristian** [comment]
Hey, update submitted. I'll see what I can do about that :)

File: composer.json [hunk 9]
```diff
@@ -16,10 +16,21 @@
...
+        "illuminate/support": "~4",
+        "phpunit/phpunit": "~4.5"
+    },
     "autoload": {
         "psr-0": {
             "Aloha\\Twilio": "src/"
         }
     },
+    "autoload-dev": {
+        "psr-0": {
```
Comment (hannesvdvreken) [inline composer.json]: psr-0 is deprecated, you can replace this with `psr-4`

File: src/Aloha/Twilio/Commands/TwilioCallCommand.php [hunk 10]
```diff
@@ -21,7 +21,7 @@ class TwilioCallCommand extends Command {
      *
      * @var string
      */
-    protected $description = 'Twilio command to test Twilio API Integration.';
+    protected $description = 'Command to test making a Call with Twilios API.';
```
Comment (hannesvdvreken) [inline src/Aloha/Twilio/Commands/TwilioCallCommand.php]: How about: "Command to make a call with Twilio."?

File: src/Aloha/Twilio/Commands/TwilioMmsCommand.php [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+     * @var string
+     */
+    protected $name = 'twilio:mms';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Command to test MMS with Twilios API.';
```
Comment (hannesvdvreken) [inline src/Aloha/Twilio/Commands/TwilioMmsCommand.php]: 'Command to send a MMS with Twilio.'

File: src/Aloha/Twilio/Commands/TwilioMmsCommand.php (hunk 6 above)
Comment (hannesvdvreken) [inline src/Aloha/Twilio/Commands/TwilioMmsCommand.php]: Please remove the constructor in this command class, I'll update the others.

**Comment by hannesvdvreken** [comment]
@CodyPChristian I tagged 1.1.0 to include this https://github.com/aloha/laravel-twilio/releases/tag/1.1.0

**Comment by CodyPChristian** [comment]
Great thank you, I will get the other things updated that you mentioned as well.

**Pull request merged**